integration-tests/paho/src/main/java/org/apache/camel/quarkus/component/paho/PahoResource.java

```diff
@@ -92,7 +116,41 @@ public String mqttExceptionDuringReconnectShouldSucceed() {
     @GET
     @Produces(MediaType.TEXT_PLAIN)
     public String readThenWriteWithFilePersistenceShouldSucceed(@QueryParam("message") String message) throws Exception {
-        producerTemplate.requestBody("paho:withFilePersistence?retained=true&persistence=FILE", message);
-        return consumerTemplate.receiveBody("paho:withFilePersistence?persistence=FILE", 5000, String.class);
+        producerTemplate.requestBody("paho:withFilePersistence?retained=true&persistence=FILE&brokerUrl=" + brokerUrl("tcp"),
+                message);
+        return consumerTemplate.receiveBody("paho:withFilePersistence?persistence=FILE&brokerUrl=" + brokerUrl("tcp"), 5000,
+                String.class);
+    }
+
+    private String brokerUrl(String protocol) {
+        return ConfigProvider.getConfig().getValue("paho.broker." + protocol + ".url", String.class);
+    }
+
+    private void setKeyStore(String keystore, String password) {
+        InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(keystore);
+
+        try {
```

Comment:
```suggestion
        try (InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(keystore)) {
```

It would be nice if we could close the InputStream